python/across_api/burstcube/api.py

```diff
@@ -1,3 +1,184 @@
 # Copyright © 2023 United States Government as represented by the
 # Administrator of the National Aeronautics and Space Administration.
 # All Rights Reserved.
+
+import gzip
+from typing import Annotated, BinaryIO, Tuple, Union
+
+from astropy.io import fits  # type: ignore
+from fastapi import Depends, File, HTTPException, Security, UploadFile, status
+
+from ..auth.api import claims, scope_authorize
+from ..base.depends import (
+    ErrorRadiusDep,
+    ExposureDep,
+    IdDep,
+    LimitDep,
+    OffsetDep,
+    OptionalDateRangeDep,
+    OptionalDurationDep,
+    OptionalRaDecDep,
+    TriggerTimeDep,
+)
+from ..base.api import app
+from .schema import BurstCubeTOORequestsSchema, BurstCubeTOOSchema, BurstCubeTriggerInfo
+from .toorequest import BurstCubeTOO, BurstCubeTOORequests
+
+
+def read_healpix_file(healpix_file: UploadFile) -> Tuple[fits.FITS_rec, str]:
+    """Read in a HEALPix file in FITS format and return the HDUList. Supports
+    gzipped FITS files"""
+
+    # Type hint for the file object
+    file: Union[gzip.GzipFile, BinaryIO]
+
+    # If the file is a gzip file, open it with gzip.open, otherwise just
+    # pass along the filehandle
+    if healpix_file.content_type == "application/x-gzip":
+        file = gzip.open(healpix_file.file, "rb")
+    else:
+        file = healpix_file.file
```

Comment:
Is this necessary? I thought that `fits.open` would transparently decompress the file for you.

Reply:
According to https://docs.astropy.org/en/stable/io/fits/#working-with-compressed-files:

> The [open()](https://docs.astropy.org/en/stable/io/fits/api/files.html#astropy.io.fits.open) function will seamlessly open FITS files that have been compressed with gzip, bzip2 or pkzip. Note that in this context we are talking about a FITS file that has been compressed with one of these utilities (e.g., a .fits.gz file).

It may be that the gzip detection is based on filename and so it doesn't work with file-like objects. Would you please check if that's the case?

Reply:
As far as I can tell looking through the code and experimenting with it, the compression detection does not work with a `BytesIO` object. It will work if you do pass it a regular `open` object of a gzipped file. Seems like maybe a bug in astropy? I can file an Issue.

Reply:
Thanks! Please add a FIXME comment with a link to the issue.

There might already be an issue for this.

Reply:
I tried to find an existing issue, but was unsuccessful, so filed one.

https://github.com/astropy/astropy/issues/16171

Added FIXME in 134a7b0.